vortex-scalar/src/lib.rs

```diff
@@ -141,6 +142,40 @@ impl Scalar {
             value: self.value,
         }
     }
+
+    pub fn default_of_type(dtype: DType) -> VortexResult<Self> {
```

Comment:
Can probably remove this and use canonical empty until we need it

Reply:
Done.